pkg/network/proxyimpl/hybrid/proxy.go

```diff
@@ -244,11 +246,11 @@ func (p *HybridProxier) switchService(name types.NamespacedName) {
 	p.switchedToUserspace[name] = p.usingUserspace[name]
 }
 
-func (p *HybridProxier) OnEndpointsAdd(endpoints *corev1.Endpoints) {
+func (p *HybridProxier) OnEndpointSliceAdd(endpoints *discoveryv1beta1.EndpointSlice) {
 	// we track all endpoints in the unidling endpoints handler so that we can succesfully
 	// detect when a service become unidling
 	klog.V(6).Infof("hybrid proxy: (always) add ep %s/%s in unidling proxy", endpoints.Namespace, endpoints.Name)
-	p.unidlingProxy.OnEndpointsAdd(endpoints)
+	p.unidlingProxy.OnEndpointSliceAdd(endpoints)
```

Comment:
I'm not familiar at all with the sdn/proxy implementation, maybe this information is redundant, but can be multiple slices for the same service, and each slice can have duplicate endpoints, kube-proxy uses a cache
https://github.com/kubernetes/kubernetes/blob/2bcbc527a760106ec89647fcf6852f37c804f4ed/pkg/proxy/endpointslicecache.go#L43-L49

Reply:
the limit of endpoints per slice is 100, so if you have more than 100 endpoints,  let's say 110 for service X you'll receive two slices Y1 and Y2, maybe with 100 and 10 endpoints each

Reply:
We really should have an e2e test in upstream that creates a service with > 100 endpoints then to exercise this.  Is there one you know of I can crib?

Reply:
Actually, for idling, wondering whether we even need to support > three or four endpoints.  The only time the user space proxy should be in play is on an idle service which has no endpoints.

Reply:
> her we even need to support > three or four endpoints.

as I said, I'm not familiar with this code, just raising some points that I think may be taking into consideration, if that is the case, it seems we should't worry about this

> We really should have an e2e test in upstream that creates a service with > 100 endpoints

this is well tested upstream

Reply:
The unidling proxy can just ignore the endpointslices and work with the endpoints like it always did. Endpoints objects always contain the full set of endpoints, even in cases where the EndpointSlice controller would start splitting things up; it just means that code working with the Endpoints objects doesn't get the efficiency wins that code working with the EndpointSlice objects would get.